docs/src/main/docs/cdi-extensions.adoc

```diff
@@ -0,0 +1,182 @@
+///////////////////////////////////////////////////////////////////////////////
+
+    Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+= Helidon CDI extensions
+:description: Helidon CDI extensions guide
+:keywords: helidon, guide, CDI
+:linkattrs:
+
+* <<Overview>>
+* <<Installing a CDI extension>>
+* <<Injecting an object using a CDI extension>>
+
+== Overview
+
+Helidon provides https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#spi[portable extensions] that you can use to inject the following objects into your Helidon MicroProfile applications:
+
+* http://brettwooldridge.github.io/HikariCP/[HikariCP data sources, window="_blank"]
+* https://github.com/xetorthio/jedis[Jedis clients, window="_blank"]
+* https://docs.cloud.oracle.com/iaas/Content/Object/Concepts/objectstorageoverview.htm[Oracle Cloud Infrastructure Object Storage clients, window="_blank"]
+
+== Installing a CDI extension
+
+The Helidon CDI extensions are available as https://mvnrepository.com/artifact/io.helidon.integrations.cdi[Maven artifacts, window="_blank"].
+
+To install an extension, ensure that the extension and its runtime dependencies are specified on your application's runtime classpath.
+
+The following examples show how to specify the required dependencies for the Helidon CDI extensions in your Maven POM (`pom.xml`).
+ 
+* To install the extension for HikariCP data sources, specify the following dependency:
+
+    <dependency>
+      <groupId>io.helidon.integrations.cdi</groupId>
+      <artifactId>helidon-integrations-cdi-datasource-hikaricp</artifactId>
+      <version>1.0.0</version>
+      <scope>runtime</scope>
+    </dependency>
+  
+* To install the extension for Jedis clients, specify the following dependency:
+
+    <dependency>
+      <groupId>io.helidon.integrations.cdi</groupId>
+      <artifactId>helidon-integrations-cdi-jedis</artifactId>
+      <version>1.0.0</version>
+      <scope>runtime</scope>
+    </dependency>
+  
+* To install the extension for Oracle Cloud Infrastructure Object Storage clients, you must first clone and install the Oracle Cloud Infrastructure Java SDK because it is not available, currently, in Maven Central.
+
+ 1. Clone the Oracle Cloud Infrastructure Java SDK artifacts.
```

Comment:
The first 2 steps describe how to build the Maven artifact from source.
Step 1 is to get the sources using git, `artifacts` here is confusing, we should mention `git` or `GitHub` instead of `artifacts`.

Reply:
I dropped 'artifacts'. But am not sure that we need to mention git or github.